plugin/src/main/java/com/denizenscript/denizen/events/block/SpongeAbsorbsScriptEvent.java

```diff
@@ -0,0 +1,59 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import org.bukkit.block.BlockState;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.SpongeAbsorbEvent;
+
+import java.util.stream.Collectors;
+
+public class SpongeAbsorbsScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // sponge absorbs
+    //
+    // @Location true
+    //
+    // @Group Block
+    //
+    // @Warning this event may in some cases double-fire, requiring usage of the 'ratelimit' command (like 'ratelimit <context.location> 1t') to prevent doubling actions.
+    //
+    // @Cancellable true
+    //
+    // @Triggers when Sponge absorbs water.
+    //
+    // @Context
+    // <context.location> returns the location of the Sponge.
+    // <context.blocks> returns a ListTag(LocationTag) of blocks that are being removed.
+    //
+    // -->
+
+    public SpongeAbsorbsScriptEvent() {
+        registerCouldMatcher("sponge absorbs");
+    }
+
+    public SpongeAbsorbEvent event;
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "location" -> new LocationTag(event.getBlock().getLocation());
+            case "blocks" -> new ListTag(event.getBlocks().stream()
+                    .map(blockState -> new LocationTag(blockState.getLocation()))
+                    .toList());
```

Comment:
I didn't mean a stream - `ListTag` has a constructor which takes in a list of _something_ and a function to convert _something_ into a denizen object; should use that